### PR TITLE
Misc validation and testing.

### PR DESCRIPTION
**Pull request by Carreau**

### CODE REVIEW

File: tests/test_npe2.py [hunk 1]
```diff
@@ -51,3 +52,40 @@ def test_cli(monkeypatch):
...
+    return data
+
+
+def _mutator_4(data):
+    assert "contributes" in data
+    data["contributes"]["commands"][0]["python_name"] = "this.has.no.colon"
+    return data
+
+
+@pytest.mark.parametrize("mutator", [_mutator_1, _mutator_2, _mutator_3, _mutator_4])
```
Comment (tlambert03) [inline tests/test_npe2.py:82]: ❤️

**Review by tlambert03 — COMMENTED** [review]
good stuff!

**Comment by Carreau** [comment]
marking as ready, it may conflicts with #3 so I'm happy to get 3 in first.

**Review by tlambert03 — APPROVED** [review]
happy to have this in now.  I'll leave it up to @nclack whether he'd prefer not having to merge or not?

File: npe2/manifest/commands.py [hunk 2]
```diff
@@ -1,3 +1,4 @@
+from textwrap import dedent
```
Comment (nclack) [inline npe2/manifest/commands.py:1]: should textwrap get added to the dependencies?

**Review by nclack — APPROVED** [review]
Would love to have this in now.

File: npe2/manifest/commands.py (hunk 2 above)
Comment (Carreau) [reply npe2/manifest/commands.py:1]: no that stdlib : https://docs.python.org/3/library/textwrap.html

Comment (nclack) [reply npe2/manifest/commands.py:1]: yay

**Pull request merged**